pkg/apis/proto/daemon/daemon.proto

```diff
@@ -73,4 +92,9 @@ service DaemonService {
   rpc GetVertexMetrics (GetVertexMetricsRequest) returns (GetVertexMetricsResponse) {
     option (google.api.http).get = "/api/v1/namespace/{namespace}/pipelines/{pipeline}/vertices/{vertex}/metrics";
   };
+
+  // GetVertexWatermark return the watermark of the given vertex based on the incoming edge buffer
+  rpc GetVertexWatermark (GetVertexWatermarkRequest) returns (GetVertexWatermarkResponse) {
+    option (google.api.http).get = "/api/v1/namespace/{namespace}/pipelines/{pipeline}/vertices/{vertex}/watermark";
```

Comment:
BTW - remove `/namespace/{namespace}` from the URI, it's not needed as the daemon service is namespaced, and the host URL already contains namespace. The `metrics` one is also removed in another PR.